presto-manager-agent/src/main/java/com/teradata/prestomanager/agent/PrestoControlUtils.java

```diff
@@ -0,0 +1,197 @@
+                        throw new PrestoManagerException("Coordinator can't be gracefully stopped");
+                    }
+                    String prestoPort = getFileProperty(PRESTO_CONFIG_FILE, "http-server.http.port");
+                    Response response = newRequestBuilder(new URI(format("localhost:%s/v1/info/state", prestoPort)))
+                            .withBody(entity((new Gson().toJson("SHUTTING_DOWN")), APPLICATION_JSON_TYPE)).buildPut().invoke();
+                    if (response.getStatus() != 200) {
+                        throw new PrestoManagerException("Failed to stop presto gracefully");
+                    }
+                }
+                catch (IOException | URISyntaxException e) {
+                    throw new PrestoManagerException("Failed to stop presto gracefully", e);
+                }
+        }
+    }
+
+    public static Response restartUsingRpm()
+    {
+        try {
+            if (!isInstalled()) {
+                return Response.status(NOT_FOUND).build();
+            }
+            int prestoRestart = executeCommand(SUBPROCESS_TIMEOUT, "service", "presto", "restart");
+            if (prestoRestart != 0) {
+                throw new PrestoManagerException("Failed to restart presto", prestoRestart);
+            }
+        }
+        catch (PrestoManagerException e) {
+            LOGGER.error(e.getMessage(), e.getCause());
+            return Response.status(INTERNAL_SERVER_ERROR).build();
+        }
+        return Response.status(OK).build();
+    }
+
+    public static Response statusUsingRpm()
+    {
+        try {
+            if (!isInstalled()) {
+                return Response.status(OK).entity(GSON.toJson("Presto is not installed"))
+                        .type(APPLICATION_JSON_TYPE).build();
+            }
+            if (!isRunning()) {
+                return Response.status(OK).entity(GSON.toJson("Presto is installed but not running"))
+                        .type(APPLICATION_JSON_TYPE).build();
+            }
+        }
+        catch (PrestoManagerException e) {
+            LOGGER.error(e.getMessage(), e.getCause());
+            return Response.status(INTERNAL_SERVER_ERROR).build();
+        }
+        try {
+            String prestoPort = getFileProperty(PRESTO_CONFIG_FILE, "http-server.http.port");
+            return newRequestBuilder(new URI(format("localhost:%s/v1/info", prestoPort))).buildGet().invoke();
+        }
+        catch (IOException | URISyntaxException e) {
+            LOGGER.error("Failed to get status", e);
+            return Response.status(INTERNAL_SERVER_ERROR).entity(GSON.toJson("Failed to get status"))
+                    .type(APPLICATION_JSON_TYPE).build();
+        }
+    }
+
+    private static boolean isInstalled()
+            throws PrestoManagerException
+    {
+        if (executeCommand("rpm", "-q", "presto-server-rpm") != 0) {
+            LOGGER.error("Presto is not installed");
```

Comment:
don't log here.  The method calling should decide whether this is an error and log if necessary